### DISCUSSION TITLE
Handling nulls in NumberInput

### DISCUSSION THREAD

**Discussion by grahampcharles** [mantinedev/mantine#4673] [Q&A] [answered]
My database (most databases?) returns optional number values as `number | null` properties. The `<NumberInput>`, however, expects `number | ''`.

So, the NumberInput fails with an understandable ```TypeError: Cannot read properties of null (reading 'toFixed')```

How is that (must-be-common?) situation typically handled? I must have to do some mapping somewhere, I assume?

**Comment by grahampcharles** [comment]
Incidentally, this is what I'm doing now. Seems kludgy, though, for such a common case.

```
    const [planOrderValue, setPlanOrderValue] = useState(props.Plan.PlanOrder);
    useEffect(() => {
        form.setFieldValue('PlanOrder', planOrderValue);
    }, [planOrderValue]);

...

    <NumberInput value={planOrderValue ?? ''}
        onChange={(val) => setPlanOrderValue(val === '' ? null : val)} />;
```

**Reply by rtivital** [reply, marked as answer]
I do not see anything wrong with this. It is just `value` and `onChange` props. You also do not need useEffect for that – you can replace `setPlanOrderValue` with `form.setFieldValue`

**Reply by grahampcharles** [reply]
Oh, sure. But then, I think, you'd need to set both the internal and display values in `onChange` -- at least that's the only way I could make it work:

```
const [planOrderValue, setPlanOrderValue] = useState<number | ''>(props.Plan.PlanOrder ?? '')
...
<NumberInput  value={planOrderValue}
                                    onChange={(val) => {
                                        setPlanOrderValue(val);
                                        form.setFieldValue("PlanOrder", val === '' ? null : val);
                                    }}
                                >
```

That's probably the simplest way?

I'm thinking I might extend `NumberInput` to add an `emptyValue` property. Does that seem ridiculous?
